core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -208,10 +207,10 @@ public synchronized void connect() throws IOException {
               + "is not able to connect to servers with SIMPLE security mode.";
           throw new UnavailableException(message, e);
         }
-        // TODO(peis): Consider closing the connection here as well.
-        if (!retryPolicy.attemptRetry()) {
-          break;
-        }
+      }
+      // TODO(peis): Consider closing the connection here as well.
+      if (!retryPolicy.attemptRetry()) {
+        break;
```

Comment:
Are there any exceptions which we should not retry?

Reply:
We will not retry the exception types, which are not `IOException` nor `TTransportException`.
Within those two types, two special cases are handled explicitly.